src/ContentRepository90/Rules/FusionNodeDepthRector.php

```diff
@@ -30,13 +30,13 @@ public function refactorFileContent(string $fileContent): string
                 '// TODO 9.0 migration: Line %LINE: You may need to rewrite "VARIABLE.depth" to Neos.Node.depth(VARIABLE). We did not auto-apply this migration because we cannot be sure whether the variable is a Node.'
             )
             ->process(fn(string $eelExpression) => preg_replace(
-                '/([a-zA-Z.]+)?(node|documentNode)\.property\\((\'|")_depth(\'|")\\)/',
-                'Neos.Node.depth($1$2)',
+                '/q\((node|documentNode)\)\.property\\((\'|")_depth(\'|")\\)/',
```

Comment:
I would rather remove the `q(...)` part at all and rely on the assumption, that `property("_depth")` can only exists on a flow query with a node.

Reply:
imo a fair assumption i guess? I mean well have to see how it behaves in client projects but yes. We can always make it more restrictive again if we feel like it. Using this regex is pretty safe anything else will definitely have to be reviewed and cannot be trusted, but i would NEVER trust these automations and ALWAYS manually review the diff and fix stuff further.

Reply:
The regex is overly complicated, those groups are not needed. And what about `site`?

```suggestion
                '/q\((site|node|documentNode)\)\.property\([\'"]_depth[\'"]\)/',
```

Also, people might be using any name instead of just `node`, `documentNode` or `site`. So why not act on `q(*).property("_depth")` instead?

```suggestion
                '/q\([^)]+\)\.property\([\'"]_depth[\'"]\)/',
```

Reply:
Yeah in this case we can be less restrictive regarding the name of the variable. For other cases where we don't know the type of the variable, we use just "site|node|documentNode" and add a comment for other occurrances.